src/main/java/io/github/fusionflux/portalcubed/framework/util/ModelUtil.java

```diff
@@ -0,0 +1,63 @@
+package io.github.fusionflux.portalcubed.framework.util;
+
+import io.github.fusionflux.portalcubed.framework.model.RenderMaterials;
+import net.fabricmc.fabric.api.renderer.v1.material.RenderMaterial;
+import net.fabricmc.fabric.api.renderer.v1.mesh.MutableQuadView;
+import net.fabricmc.fabric.api.renderer.v1.mesh.QuadEmitter;
+import net.fabricmc.fabric.api.renderer.v1.render.RenderContext;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.model.BakedModel;
+import net.minecraft.core.Direction;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.block.state.BlockState;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+public class ModelUtil {
+	private static final Direction[] FACES = Arrays.copyOf(Direction.values(), 7);
+
+	public static TextureAtlasSprite getSprite(ResourceLocation texture) {
+		return Minecraft.getInstance().getTextureAtlas(TextureAtlas.LOCATION_BLOCKS).apply(texture);
+	}
+
+	public static void normalizeUV(MutableQuadView quad, TextureAtlasSprite sprite) {
+		float uMin = sprite.getU0();
+		float uSpan = sprite.getU1() - uMin;
+		float vMin = sprite.getV0();
+		float vSpan = sprite.getV1() - vMin;
+		for (int i = 0; i < 4; i++) {
+			quad.uv(i, (quad.u(i) - uMin) / uSpan, (quad.v(i) - vMin) / vSpan);
+		}
+	}
+
+	public static void emitVanillaQuads(
```

Comment:
this is bad, vanilla baked quads should be avoided. getQuads could crash on a subclass of SimpleBakedModel

Reply:
`getQuads` is what indigo and indium call internally whenever you do `super.emit[Block/Item]Quads` on a WrapperBakedModel/ForwardingBakedModel. this `emitVanillaQuads` is a modified version of what indigo does internally, I don't see any problem with this.

Reply:
look at `VanillaModelEncoder` called from the defaults of FabricBakedModel

Reply:
those are the default impls, mods don't have to use them. If `isVanillaAdapter` is false, calling getQuads is invalid.
In theory a mod could extend SimpleBakedModel so it passes the instanceofs but then throw in getQuads.

Reply:
What do you suggest instead of `getQuads`

Reply:
emitBlock/ItemQuads

Reply:
That doesn't allow me to alter the FRAPI quad based on the data in the baked quad